Keas.Mvc/Services/IdentityService.cs

```diff
@@ -228,8 +219,38 @@ public async Task<User> GetByKerberos(string kerb)
 
 
             return rtValue;
-        }
-
+        }
+
+        private async Task<string> GetEmailForUser(KerberosResult ucdKerbPerson, IetClient clientws)
+        {
+            if(!string.IsNullOrWhiteSpace(ucdKerbPerson.CampusEmail))
+            {
+                //This is the preferred email
+                return ucdKerbPerson.CampusEmail;
+            }
+            var ucdContactResult = await clientws.Contacts.Get(ucdKerbPerson.IamId);
+
+            if (ucdContactResult.ResponseData.Results.Length == 0)
+            {
+                //So if there are no contact details, we will return null instead of trying to fake the email. This matches up with current processing, but maybe it is wrong?
+                return null;
+            }
+
+            if (ucdContactResult.ResponseData.Results.Length > 0)
+            {
+                var ucdContact = ucdContactResult.ResponseData.Results.Where(a => a.Email != null).FirstOrDefault();
+                if (ucdContact != null)
+                {
+                    //Or try to get the contact email
+                    return ucdContact.Email;
+                }
+            }
+
+            //Or just use the kerb with email. Might be ok.
+            return $"{ucdKerbPerson.UserId}@ucdavis.edu";
```

Comment:
is this something we normally do? this isn't a safe bet. my kerb@ucdavis.edu would be wrong